google/cloud/pubsub/internal/defaults_test.cc

```diff
@@ -27,13 +27,11 @@ using ms = std::chrono::milliseconds;
 
 TEST(OptionsTest, PublisherDefaults) {
   auto opts = DefaultPublisherOptions(Options{});
-  EXPECT_EQ(ms(10), opts.get<pubsub::MaxHoldTimeOption>());
-  EXPECT_EQ(100, opts.get<pubsub::MaxBatchMessagesOption>());
-  EXPECT_EQ(1024 * 1024L, opts.get<pubsub::MaxBatchBytesOption>());
-  EXPECT_EQ((std::numeric_limits<std::size_t>::max)(),
-            opts.get<pubsub::MaxPendingBytesOption>());
-  EXPECT_EQ((std::numeric_limits<std::size_t>::max)(),
-            opts.get<pubsub::MaxPendingMessagesOption>());
+  EXPECT_LT(0, opts.get<pubsub::MaxHoldTimeOption>().count());
```

Comment:
Actually, the default values here matter, because there are common standards across all Pub/Sub libraries we want to follow.

Reply:
This defaults test copied checks from `publisher_options_test.cc`.
https://github.com/googleapis/google-cloud-cpp/blob/0eafcc84660df9070e08dbdf6807a91d6a8efa65/google/cloud/pubsub/publisher_options_test.cc#L25-L26
https://github.com/googleapis/google-cloud-cpp/blob/0eafcc84660df9070e08dbdf6807a91d6a8efa65/google/cloud/pubsub/publisher_options_test.cc#L47-L48
https://github.com/googleapis/google-cloud-cpp/blob/0eafcc84660df9070e08dbdf6807a91d6a8efa65/google/cloud/pubsub/publisher_options_test.cc#L58-L59
etc.

Should I also update the original test to check for equality of default values?

What about `subscriber_options_test.cc`?
https://github.com/googleapis/google-cloud-cpp/blob/0eafcc84660df9070e08dbdf6807a91d6a8efa65/google/cloud/pubsub/subscriber_options_test.cc#L31-L37

Reply:
We only need one test that verifies the defaults match the defaults in other client libraries.  If we have one test that checks `default == specific-value` I am happy.  If all our tests amount to `default != 0` then I am not happy.

Does that make sense?

Reply:
It does make sense. It implies that you have been unhappy with the tests for a while ;)